### PR TITLE
Fix - No GA code when `Enable Enhanced eCommerce` option is active but WooCommerce is not active

### LINKED ISSUE #10091: WooCommerce Analytics: do not output the tracking code in a corner case
Jetpack does not output the GA tracking code under this condition: 

* `Enable Enhanced eCommerce` option is active
* But WooCommerce is not active

#### Steps to reproduce the issue

1. Activate WooCommerce on a Premium or Pro site.
2. In https://wordpress.com/settings/traffic/my-site.com, activate `Enable Enhanced eCommerce`.
3. Deactivate WooCommerce 
4. Visit the site without login as an admin and check the HTML output. 

#### What I expected

A proper Google Analytics tracking code is created by Jetpack (we can search with `Jetpack Google Analytics`).  

#### What happened instead

No Google Analytics code.

#### Debug further

The error is around these lines:

1. https://github.com/Automattic/jetpack/blob/20f2dfb1816c158778a856ed3115a0ce96651ed5/modules/google-analytics/classes/wp-google-analytics-universal.php#L54-L57
2. https://github.com/Automattic/jetpack/blob/20f2dfb1816c158778a856ed3115a0ce96651ed5/modules/google-analytics/wp-google-analytics.php#L52-L59

(Originally reported in 2580-gh-jpop-issues)

### PR DESCRIPTION
**Pull request by htdat**
👋from a HE. My first PR. Thanks for the feedback in advance :) 

Fixes #10091

#### Changes proposed in this Pull Request:

Currently, Jetpack does not output the GA tracking code in a specific case, which can be replicated as you see on #10091 : 

* `Enable Enhanced eCommerce` option is active
* But WooCommerce is not active

#### Testing instructions:

1. Activate WooCommerce on a Premium or Pro site.
2. In https://wordpress.com/settings/traffic/my-site.com, activate Enable Enhanced eCommerce.
3. Deactivate WooCommerce
4. Visit the site without login as an admin. 
5. Check the HTML output and find the GA code by searching `Jetpack Google Analytics`.

#### Proposed changelog entry for your changes:

Fix Bug - Do not output the Google Analytics code when `Enable Enhanced eCommerce` option is active but WooCommerce is not active

### CODE REVIEW

File: modules/google-analytics/wp-google-analytics.php [hunk 1]
```diff
@@ -50,7 +50,9 @@ class Jetpack_Google_Analytics {
 	 * @return void
 	 */
 	private function __construct() {
-		if ( Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+		// At this time, we only leverage universal analytics when enhanced ecommerce is selected and WooCommerce is active.
+		// Otherwise, don't bother emitting the tracking ID or fetching analytics.js
+		if ( Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() && class_exists( 'WooCommerce' ) ) {
```
Comment (kraftbj) [inline modules/google-analytics/wp-google-analytics.php]: This is extremely nitpicky, but sharing to help frame something to keep in mind. With an `&&` operation, it checks left-to-right and fails on the first failed condition—so the `enhanced_ecommerce_tracking_is_enabled`, then the `class_exists` check. The `class_exists` check is pretty cheap, but `enhanced_ecommerce_tracking_is_enabled` eventually calls `get_option`, which may or may not be a db hit (didn't look to see if the option is autoloaded or not).

In this case, it probably isn't a huge thing, but for the sake of performance, thinking about the order things are checked can help you out.

Comment (htdat) [reply modules/google-analytics/wp-google-analytics.php]: That makes sense to me! Thanks Kraft 👍

I've pushed a new commit to reorder the condition positions

**Comment by allendav** [comment]
Hi @htdat - what's the use case that this addresses? Thx!

**Comment by kraftbj** [comment]
@allendav The use case I'm testing for is someone who had a WooCommerce site and started tracking with enhanced eCommerce, then stopped using WooCommerce (for whatever reason). Before the PR, it would disable all Google Analytics code since WooCommerce wasn't active.

File: modules/google-analytics/classes/wp-google-analytics-universal.php [hunk 2]
```diff
@@ -49,12 +49,6 @@ public function wp_head() {
 			return;
 		}
 
-		// At this time, we only leverage universal analytics for enhanced ecommerce. If WooCommerce is not
-		// present, don't bother emitting the tracking ID or fetching analytics.js
-		if ( ! class_exists( 'WooCommerce' ) ) {
```
Comment (allendav) [inline modules/google-analytics/classes/wp-google-analytics-universal.php]: Why remove this? Although the other change you made will prevent this class from being instantiated unless WooCommerce is active, this is a nice extra layer of protection that avoids this class becoming dependent on another class for sanity.

File: modules/google-analytics/wp-google-analytics.php [hunk 3]
```diff
@@ -50,7 +50,9 @@ class Jetpack_Google_Analytics {
 	 * @return void
 	 */
 	private function __construct() {
-		if ( Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+		// At this time, we only leverage universal analytics when enhanced ecommerce is selected and WooCommerce is active.
+		// Otherwise, don't bother emitting the tracking ID or fetching analytics.js
```
Comment (allendav) [inline modules/google-analytics/wp-google-analytics.php]: Maybe reword or even completely remove this comment because it is a little confusing since the else condition will emit a tracking ID ala legacy.

**Comment by allendav** [comment]
@kraftbj wrote

> @allendav The use case I'm testing for is someone who had a WooCommerce site and started tracking with enhanced eCommerce, then stopped using WooCommerce (for whatever reason). Before the PR, it would disable all Google Analytics code since WooCommerce wasn't active.

OK, fair enough. So we want to "fall back" to legacy analytics if WooCommerce is deactivated while this "enhanced eCommerce" option is active - that's cool.
